ONSdigital/ssdc-rm-support-tool
# Production harden test/dummy user security

# Motivation and Context
Without IAP, testing local security requires some mechanisms to spoof user identity. We should ensure that the testing code and config can't be used in an attack, or be accidentally left in production config.

# What has changed
The environment variable `DUMMYUSERIDENTITY-ALLOWED` has to be set equal to true to allow the dummy testing user. Naturally, it defaults to false.

# How to test?
Change `DUMMYUSERIDENTITY-ALLOWED` to be false in docker dev and GCP and check that user access is totally forbidden for requests which don't go via IAP... e.g. try `curl -i http://support-tool:80/api/whoami` from the support tool in GCP.

# Links
Trello: https://trello.com/c/7FMv2msJ

# Screenshots (if appropriate):
<img width="656" alt="Screenshot 2021-10-18 at 14 18 59" src="https://user-images.githubusercontent.com/41681172/137741616-64a23271-4442-47c4-8dcc-b891e11372b2.png">
